iterative/resource_task.go

```diff
@@ -159,6 +163,14 @@ func resourceTask() *schema.Resource {
 }
 
 func resourceTaskCreate(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	logger := utils.TpiLogger(d)
+	logger.Info(fmt.Sprintf(logTpl, "Creation", "Creation"))
+
+	spot := d.Get("spot").(float64)
+	if spot > 0 {
+		logger.Info("Fixed spot price does not ensure the Task to be launched. It's recommended to use auto spot price (0)")
+	}
```

Comment:
```suggestion
		if spot < 0.05 {
			logger.Warn("Maximum spot price likely too low. Try at least `spot=0.05` USD/h or auto (`spot=0`).")
		}
	}
```

Reply:
How do we know 0.05 is effective? Probably you wont never have a GPU under $1/h and we should warn about that possibility. Thats why  tried to warn the issue of setting a fixed price vs auto-bid

Reply:
💯 Setting arbitrary thresholds is rather useless. Warnings should be printed for either `spot > 0` or `spot < cloud.CurrentSpotBiddingPrice()` which isn't implemented nor particularly trivial to implement.

Reply:
**TL;DR** Let's keep `spot > 0` as the warning condition.

Reply:
lets see what @casperdcl thinks

Reply:
`spot > 0` is already warned about.
There should be an additional warning if spot is also `< something_ridiculous`.

I think `0.02` is a reasonable bare minimum based on `t2.micro`.

We could do `gpu ? 1 : 0.02` or even:

type | minimum recommended price
---|---
m | 0.45
l | 1.75
xl | 3.60
m+k80 | 1.10
l+k80 | 8.70
xl+k80 | 17.30
m+v100 | 1.80
l+v100 | 14.70
xl+v100 | 29.40

Reply:
> _There should be an additional warning if spot is also `< something_ridiculous`_

Spot bidding prices are different across providers, and fluctuate considerably. Hardcoding what is and what isn't “ridiculous” for every instance offered by every cloud providers doesn't sound like a practical improvement.

![Captura de pantalla 2022-04-01 a las 22 37 13](https://user-images.githubusercontent.com/11387611/161337926-ae7c452c-6f7e-4e74-9afa-57c75918cb04.png)

Reply:
I'd rather rely on the `spot > 0` message and warn users: if they set a fixed price, they must know very well what they're doing.